qiskit/circuit/gate.py

```diff
@@ -237,5 +237,11 @@ def validate_parameter(self, parameter):
                 3,
             )
             return parameter
+        elif isinstance(parameter, complex):
+            if np.isclose(np.imag(parameter), 0):
+                return np.real(parameter)
```

Comment:
This seems dangerous, and I'm not a huge fan.  The tolerance for conversion to `real` is implicit, so if your parameters are _all_ small values, then this would silently fail on any small complex.  I think the code that actually wants to set the parameter should be responsible for this conversion, if they want it, and `complex` should raise the "unsupported type" error even if it's like `(1 + 0j)`.